week-3/alarmclock/alarmclock.js

```diff
@@ -1,4 +1,54 @@
-function setAlarm() {}
+let intervalId;
+let bgIntervalId;
+
+function setAlarm() {
+  clearInterval(intervalId);
+  clearInterval(bgIntervalId);
+  setBackground("none");
+  document.getElementById("alarmSet").classList.remove("invalidInput");
+
+  let alarmSet = document.getElementById("alarmSet").value;
+
+  if (alarmSet > 3599) {
+    document.getElementById("alarmSet").classList.add("invalidInput");
+    return;
+  }
+
+  setTime(alarmSet);
+
+  intervalId = setInterval(function () {
+    alarmSet = alarmSet - 1;
+    setTime(alarmSet);
+    if (alarmSet == 0) {
+      setBackground("yellow", "white");
+      playAlarm();
+      clearInterval(intervalId);
+    }
+  }, 1000);
+}
+
+function setBackground(color1, color2) {
+  let i = 0;
+
+  bgIntervalId = setInterval(function () {
+    if (i % 2 == 0) {
+      document.body.style.background = color1;
+    } else {
+      document.body.style.background = color2;
+    }
+    i++;
+  }, 500);
+}
+
+function setTime(time) {
+  document.getElementById("time").innerText = toTime(time);
+}
+
+function toTime(seconds) {
+  var date = new Date(null);
```

Comment:
I really like what you've done here. I got a bit confused about this, forgot that new Date(null) gives a date where the secconds are always zero. I probaly would have added a comment here to make that super clear but that's something for next time and I might be biased

Reply:
To be honest, I found this function from stackoverflow to convert seconds to time string with format hh:mm:ss. :)

Reply:
Fair enough, all the more reason to add a comment to ensure that you get it :)